api/src/pcapi/core/users/email/update.py

```diff
@@ -162,19 +166,28 @@ def request_email_update_from_pro(user: models.User, email: str, password: str)
     send_pro_user_emails_for_email_change(user, email, expiration_date)
 
 
-def check_and_increment_email_update_attempts_count(user: models.User) -> None:
+def check_email_update_attempts_count(user: models.User) -> None:
     """Check if the user has reached the maximum number of email update attempts.
     If yes, raise an exception.
-    The number of attempts is *always* incremented, even if the limit is reached.
     """
     update_email_attempts_key = f"update_email_attemps_user_{user.id}"
-    count = app.redis_client.incr(update_email_attempts_key)  # type: ignore [attr-defined]
+    if (
+        app.redis_client.exists(update_email_attempts_key)  # type: ignore [attr-defined]
+        and int(app.redis_client.get(update_email_attempts_key)) >= settings.MAX_EMAIL_UPDATE_ATTEMPTS  # type: ignore [attr-defined]
+    ):
+        raise exceptions.EmailUpdateLimitReached()
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** [‏DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
```python
    update_email_attempts = app.redis_client.get(update_email_attempts_key)
    if update_email_attempts and int(update_email_attempts) >= settings.MAX_EMAIL_UPDATE_ATTEMPTS:
        raise exceptions.EmailUpdateLimitReached()
```